src/core/Akka.TestKit/TestKitBase_Receive.cs

```diff
@@ -523,34 +538,38 @@ public IReadOnlyList<T> ReceiveWhile<T>(Func<object, T> filter, TimeSpan? max =
             while (count < msgs)
             {
                 // Peek the message on the front of the queue
```

Comment:
Add CancellationToken check before this comment, throw an exception that Task is cancelled using the ThrowIf method.